kyuubi-common/src/main/scala/org/apache/kyuubi/service/TFrontendService.scala

```diff
@@ -159,8 +159,10 @@ abstract class TFrontendService(name: String)
     res.setServerProtocolVersion(protocol)
     val userName = getUserName(req)
     val ipAddress = getIpAddress
-    val configuration = Map(KYUUBI_CLIENT_IP_KEY -> ipAddress) ++
-      Option(req.getConfiguration).map(_.asScala.toMap).getOrElse(Map.empty[String, String])
+    val configuration =
+      Map(KYUUBI_CLIENT_IP_KEY -> ipAddress) ++
+        Option(req.getConfiguration).map(_.asScala.toMap).getOrElse(Map.empty[String, String]) ++
+        Map(KYUUBI_SESSION_CONNECTION_URL_KEY -> connectionUrl)
```

Comment:
re-order the KYUUBI_SESSION_CONNECTION_URL_KEY to prevent it be overwritten by client.